File: src/protocol/AutomaticBlindsCommand.cpp
#include "AutomaticBlindsCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;

AutomaticBlindsCommand::AutomaticBlindsCommand()
    : Command(10, CommandType::AUTOMATIC_BLINDS)
{}

AutomaticBlindsCommand::~AutomaticBlindsCommand()
{}

std::vector<uint8_t> AutomaticBlindsCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

std::string AutomaticBlindsCommand::getName()
{
    return string("BlindsStatusCommand");
}

void AutomaticBlindsCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void AutomaticBlindsCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/AutomaticBlindsCommand.java
package wpam.mobile_client.protocol;

import java.util.ArrayList;

public final class AutomaticBlindsCommand extends Command
	{
		public AutomaticBlindsCommand()
		{
			super(10, CommandType.AUTOMATIC_BLINDS);
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(commandType_.getValue());

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("BlindsStatusCommand");
		}
		@Override
		public void accept(CommandVisitor visitor)
		{
			visitor.visit(this);
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(1);
		}
	}